app/models/survey_response.rb

```diff
@@ -0,0 +1,14 @@
+class SurveyResponse < ApplicationRecord
+  belongs_to :feedback
+  belongs_to :survey_question
+
+  validates_presence_of :value, if: ->{ survey_question.required? }
+
+  before_create :copy_prompt_from_question
+
+  private
+
+  def copy_prompt_from_question
+    self.prompt = survey_question.prompt
```

Comment:
Could we delegate prompt to survey_question, or does this need to be saved to the response itself?

Reply:
The response saves both the question that was asked and the answer. This is in case the survey gets edited later and the question changes. It's a bit of a speculative concern :man_shrugging:. Instead I could have made the question prompts non-editable which would have made sense too. I'm not sure whether one approach is better.

Reply:
Ahh got it. In that case I think this is better. Not being able to edit a small typo sounds frustrating.